### PR TITLE
refactor: Refactor `data-plane-transfer`

### LINKED ISSUE #2274: Refactor `data-plane-transfer`
# Feature Request

Some improvement have to be done in the code of `data-plane-transfer`.

## Which Areas Would Be Affected?
Control Plane, DPF

## Why Is the Feature Desired?
Simplify code and provide better documentation.

## Solution Proposal
- Move the `DataPlaneClient` and its implementations (remote, embedded) into the `data-plane` module (as it is already done for Data Plane selector)
- Merge `data-plane-transfer-sync` (whose name was confusing) and `data-plane-transfer-client` into a single module called `data-plane-transfer`
- Ease test/demo setup by providing an automatically generated key pair for signing/verifying tokens
- Provide proper UML flow diagrams for both Client Pull and Provider Push data transfers.

### PR DESCRIPTION
**Pull request by bscholtes1A**
## What this PR changes/adds

Refactor `data-plane-transfer`:

- Move the `DataPlaneClient` and its implementations (remote, embedded) into the `data-plane` module (as it is already done for Data Plane selector)
- Merge `data-plane-transfer-sync` (whose name was confusing) and `data-plane-transfer-client` into a single module called `transfer-data-plane`
- Ease test/demo setup by providing an automatically generated key pair for signing/verifying tokens if none is provided from the config
- Provide proper UML flow diagrams for both Client Pull and Provider Push data transfers.

## Why it does that

Simplify code management and improve documentation.

## Linked Issue(s)

Closes #2274 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

File: extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/flow/DataPlaneTransferDataFlowController.java [hunk 1]
```diff
@@ -24,44 +24,41 @@
...
-        if (!StringUtils.isNullOrBlank(type)) {
-            return !HTTP_PROXY.equals(dataRequest.getDestinationType());
-        }
-        return false;
+        return true;
     }
 
     @Override
     public @NotNull StatusResult<Void> initiateFlow(DataRequest dataRequest, DataAddress contentAddress, Policy policy) {
+        if (HTTP_PROXY.equals(dataRequest.getDestinationType())) {
```
Comment (ndr-brt) [inline extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/flow/DataPlaneTransferDataFlowController.java:52]: I preferred when they were two different `DataFlowController`s: one for `ProviderPush` and one for `ConsumerPull`, it showed the difference in a clearer way in my opinion.

File: extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/security/KeyPairWrapperImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.transfer.security;
+
+import org.eclipse.edc.connector.dataplane.transfer.spi.security.KeyPairWrapper;
+
+import java.security.KeyPair;
+
+public class KeyPairWrapperImpl implements KeyPairWrapper {
```
Comment (ndr-brt) [inline extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/security/KeyPairWrapperImpl.java:21]: The name could refer to the `ConsumerPull` transfer type, like `ConsumerPullTransferKeyPair`.

File: extensions/data-plane/data-plane-api/src/main/java/org/eclipse/edc/connector/dataplane/api/controller/DataPlanePublicApiController.java [hunk 3]
```diff
@@ -50,13 +50,13 @@
 public class DataPlanePublicApiController implements DataPlanePublicApi {
 
     private final DataPlaneManager dataPlaneManager;
-    private final TokenValidationClient tokenValidationClient;
+    private final DataPlaneDataAddressResolver tokenValidationClient;
```
Comment (ndr-brt) [inline extensions/data-plane/data-plane-api/src/main/java/org/eclipse/edc/connector/dataplane/api/controller/DataPlanePublicApiController.java:53]: the field should be called like `dataAddressResolver`

File: extensions/data-plane/data-plane-api/src/main/java/org/eclipse/edc/connector/dataplane/api/validation/TokenValidationClientImpl.java [hunk 4]
```diff
@@ -27,7 +27,7 @@
 
 import static java.lang.String.format;
 
-public class TokenValidationClientImpl implements TokenValidationClient {
+public class TokenValidationClientImpl implements DataPlaneDataAddressResolver {
```
Comment (ndr-brt) [inline extensions/data-plane/data-plane-api/src/main/java/org/eclipse/edc/connector/dataplane/api/validation/TokenValidationClientImpl.java:30]: The name of the class should change accordingly

File: spi/data-plane/data-plane-spi/src/main/java/org/eclipse/edc/connector/dataplane/spi/resolver/DataPlaneDataAddressResolver.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+package org.eclipse.edc.connector.dataplane.spi.resolver;
+
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+
+/**
+ * Resolves the access token received in input of Data Plane public API into a {@link DataAddress}.
+ */
+@FunctionalInterface
+public interface DataPlaneDataAddressResolver {
```
Comment (ndr-brt) [inline spi/data-plane/data-plane-spi/src/main/java/org/eclipse/edc/connector/dataplane/spi/resolver/DataPlaneDataAddressResolver.java:24]: The interface could be called `DataAddressResolver` and the implementation `DataPlaneDataAddressResolver`, or, maybe better `ConsumerPullTransferDataAddressResolver`

File: extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/DataPlaneTransferExtension.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    @Inject
+    private DataPlaneClient dataPlaneClient;
+
+    @Inject
+    private ControlApiConfiguration controlApiConfiguration;
+
+    @Inject
+    private KeyPairWrapper keyPairWrapper;
+
+    @Inject(required = false)
```
Comment (ndr-brt) [inline extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/DataPlaneTransferExtension.java]: Should this be mandatory? a Provider push dataflow controller without the control plane api url is a dataflow where the transfer can never succeed.

**Review by ndr-brt — COMMENTED** [review]
I think we're almost ready to go.
I was wondering if the `data-plane-transfer` could be called `transfer-data-plane` instead, do differentiate it from the `data-plane` extensions, doing like this it will solve part of #2229

File: extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/DataPlaneTransferExtension.java (hunk 6 above)
Comment (bscholtes1A) [reply extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/DataPlaneTransferExtension.java]: Let's tackle it separately as PR is already pretty big ;)

**Review by wolf4ood — APPROVED** [review]

**Comment by bscholtes1A** [comment]
> I think we're almost ready to go. I was wondering if the `data-plane-transfer` could be called `transfer-data-plane` instead, do differentiate it from the `data-plane` extensions, doing like this it will solve part of #2229

Done as per last commit ;)

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**